lib/circuitbox/circuit_breaker.rb

```diff
@@ -77,6 +77,18 @@ def run(circuitbox_exceptions: true)
       response
     end
 
+    # Starts a circuit run block and returns circuit_open state, and a completion 
+    # that can be used to complete the run execution if the circuit is not already opened.
+    # @return circuit_open?, completion(success: true|false)
+    def run_and_get_completion
```

Comment:
We will have to modify the timer classes to support this pattern as well if we want the timing gauge to be supported.